### PR TITLE
Improve assignable roles and add ?create_role

### PR DESCRIPTION
**Pull request by davidlougheed**
hehe

### CODE REVIEW

**Review by le-potate — APPROVED** [review]
good changes, will make it easier to convert this to database eventually, especially by moving stuff that mods might want to change like categories and limits to globals which should probably become db later. Using ?iam and ?iamn makes the commands way more intuitive than having to write `add` after the command.

File: cogs/roles.py [hunk 1]
```diff
@@ -17,129 +15,228 @@
...
+        "pronouns",
+        "fields",
+        "faculties",
+        "years",
+        "generics",
+    )
+
+    CATEGORY_LIMITS = {
+        "pronouns": None,
+        "fields": 2,
```
Comment (krubenok) [inline cogs/roles.py:45]: Limit here should be 3 IMO. Arts is major + 2 minors.

Comment (davidlougheed) [reply cogs/roles.py:45]: 5c27cd3eb5efe0b0e56c02415ac994957a49cfea

**Comment by le-potate** [comment]
Related to #136

**Pull request merged**